generators/pkg/writer/write.go

```diff
@@ -26,28 +27,31 @@ func WriteToDisk(filename string, writer io.Writer) error {
 		return fmt.Errorf("unable to parse the file from %s: %w", filename, err)
 	}
 
+	// resrouces needed to write to disk
 	task := generator.GenerateTask(github)
 	pipeline, err := generator.GeneratePipeline(github)
-
-	// write task
-	if _, err := writer.Write([]byte("---\n")); err != nil {
-		return err
-	}
-	data, err := yaml.Marshal(&task)
 	if err != nil {
-		return fmt.Errorf("unable to marshal the task %s: %w", task.Name, err)
+		return fmt.Errorf("unable to get the pipeline: %w", err)
 	}
-	if _, err := writer.Write(data); err != nil {
-		return err
+	trigger := generator.GenerateTrigger(pipeline)
+
+	// pack up all the objects
+	pac := []runtime.Object{task, pipeline, trigger.TriggerBinding, trigger.TriggerTemplate, trigger.EventListener}
+	for _, o := range pac {
```

Comment:
Should range over the objects in the write method instead?

This looks like it might be a good candidate for a [variadic function](https://golang.org/ref/spec#Passing_arguments_to_..._parameters). e.g. 

```go
func writeYaml(writer io.Writer, objs ...runtime.Object) error {
  for _, o := range objs {
    ...
}